pkg/resources/saml2_integration.go

```diff
@@ -0,0 +1,543 @@
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Metadata created by Snowflake to provide to SAML2 provider.",
+	},
+	"saml2_digest_methods_used": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Digest methods used for this integration.",
+	},
+	"saml2_signature_methods_used": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Signature methods used for this integration.",
+	},
+	"allowed_user_domains": {
+		Type: schema.TypeList,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+		Optional:    true,
+		Description: "A list of email domains that can authenticate with a SAML2 security integration.",
+	},
+	"allowed_email_patterns": {
+		Type: schema.TypeList,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+		Optional:    true,
+		Description: "A list of regular expressions that email addresses are matched against to authenticate with a SAML2 security integration.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the integration.",
+	},
+	"created_on": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Date and time when the SAML2 integration was created.",
+	},
+}
+
+// SAML2Integration returns a pointer to the resource representing a SAML2 security integration.
+func SAML2Integration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextSAMLIntegration,
+		ReadContext:   ReadContextSAMLIntegration,
+		UpdateContext: UpdateContextSAMLIntegration,
+		DeleteContext: DeleteContextSAMLIntegration,
+		CustomizeDiff: customdiff.Sequence(
+			SetEmptyForceNewIfChange[any]("allowed_user_domains"),
+			SetEmptyForceNewIfChange[any]("allowed_email_patterns"),
+			SetEmptyForceNewIfChangeString("saml2_sp_initiated_login_page_label"),
+		),
+		Schema: saml2IntegrationSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func CreateContextSAMLIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	issuer := d.Get("saml2_issuer").(string)
+	ssoUrl := d.Get("saml2_sso_url").(string)
+	providerRaw := d.Get("saml2_provider").(string)
+	cert := d.Get("saml2_x509_cert").(string)
+	provider, err := sdk.ToSaml2SecurityIntegrationSaml2ProviderOption(providerRaw)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	req := sdk.NewCreateSaml2SecurityIntegrationRequest(id, issuer, ssoUrl, provider, cert)
+
+	if !d.GetRawConfig().AsValueMap()["enabled"].IsNull() {
+		req.WithEnabled(d.Get("enabled").(bool))
+	}
+
+	if v, ok := d.GetOk("saml2_sp_initiated_login_page_label"); ok {
+		req.WithSaml2SpInitiatedLoginPageLabel(v.(string))
+	}
+
+	req.WithSaml2EnableSpInitiated(d.Get("saml2_enable_sp_initiated").(bool))
```

Comment:
it is optional, so why we don't check it? (the same with some others)

Reply:
Fixed.